adding basic+jwt auth methods to post/presign sinks

miki725 opened the pull request:
this will allow to add jwt auth type for the console sinks very easily and still be able to reuse them elsewhere in the codebase

viega reviewed (approved):
I found the implementation somewhat confusing, took me a while to work through.

1) No documentation. Pretty much everything in nimutils does have doc strings for the public functions at this point.

2) For 150 lines of code, it's surprisingly spread out. I would merge into one module, and make it clear that it's related to the pub/sub API. I think *that* maybe should go into its own sub-directory soon (doesn't have to be now), but I'd move everything in auth, jwt and most of the changes in pubsub.nim into a single module.
 
Not sure this needs to be fixed right away, but shouldn't leave it for long if so.

miki725 commented:
most of the implementation was based on how sinks were implemented with similar registration flow in there

sure can clean up the modules in later PR. thanks

miki725 commented:
without refactoring too much here is new state:

* moved all auth logic out of pubsub and moved it to `auth.nim`
* left jwt.nim standalong as it an independent utility
* we can merge pubsub/sinks in the future as its confusing whats where right now as they overap however dont want to expand the scope of just adding auth implementation in this PR
